File: experiments_records/1_latent vulnerabilities/1_project_level/php/educollab_php_v0_iteration4/educollab_php_v0/src/repos/UserRepo.php
<?php

class UserRepo {
    public function __construct(private PDO $pdo) {}

    public function create(string $username, string $password, string $displayName): array {
        $hash = password_hash($password, PASSWORD_DEFAULT);
        $stmt = $this->pdo->prepare('INSERT INTO users(username, password_hash, display_name, created_at, is_global_admin) VALUES (?,?,?,?,0)');
        $stmt->execute([$username, $hash, $displayName, now_iso()]);
        return $this->getById((int)$this->pdo->lastInsertId());
    }

    public function getById(int $userId): ?array {
        $stmt = $this->pdo->prepare('SELECT user_id, username, display_name, created_at, is_global_admin FROM users WHERE user_id=?');
        $stmt->execute([$userId]);
        $row = $stmt->fetch();
        if (!$row) return null;
        $row['is_global_admin'] = !empty($row['is_global_admin']);
        return $row;
    }

    public function getByUsername(string $username): ?array {
        $stmt = $this->pdo->prepare('SELECT user_id, username, display_name, created_at, is_global_admin FROM users WHERE username=?');
        $stmt->execute([$username]);
        $row = $stmt->fetch();
        if (!$row) return null;
        $row['is_global_admin'] = !empty($row['is_global_admin']);
        return $row;
    }

    public function getAuthByUsername(string $username): ?array {
        $stmt = $this->pdo->prepare('SELECT * FROM users WHERE username=?');
        $stmt->execute([$username]);
        $row = $stmt->fetch();
        if (!$row) return null;
        $row['is_global_admin'] = !empty($row['is_global_admin']);
        return $row;
    }

    public function listAll(): array {
        $stmt = $this->pdo->query('SELECT user_id, username, display_name, created_at, is_global_admin FROM users ORDER BY user_id ASC');
        $rows = $stmt->fetchAll();
        foreach ($rows as &$row) {
            $row['is_global_admin'] = !empty($row['is_global_admin']);
        }
        return $rows;
    }
}

File: EduCollab_benchmark/2_requirement_level_generated_artifacts/python/educollab_flask_req05/models/user_repo.py
from models.db import get_db


def create_user(username, password_hash, display_name):
    db = get_db()
    cursor = db.execute(
        'INSERT INTO users (username, password_hash, display_name) VALUES (?, ?, ?)',
        (username, password_hash, display_name),
    )
    db.commit()
    return get_user_by_id(cursor.lastrowid)


def get_user_by_username(username):
    db = get_db()
    return db.execute(
        'SELECT user_id, username, password_hash, display_name, created_at FROM users WHERE username = ?',
        (username,),
    ).fetchone()


def get_user_by_id(user_id):
    db = get_db()
    return db.execute(
        'SELECT user_id, username, password_hash, display_name, created_at FROM users WHERE user_id = ?',
        (user_id,),
    ).fetchone()


def list_users():
    db = get_db()
    return db.execute(
        'SELECT user_id, username, display_name, created_at FROM users ORDER BY user_id ASC'
    ).fetchall()
